src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,615 @@
+							if (!superScrollView.ContentOffset.Equals(newContentOffset))
+							{
+								if (View?.Superview is UIStackView)
+									superScrollView.SetContentOffset(newContentOffset, UIView.AnimationsEnabled);
+								else
+									superScrollView.ContentOffset = newContentOffset;
+							}
+						}
+						superScrollView = superScrollView.FindResponder<UIScrollView>();
+					}
+				});
+			}
+
+			if (IsKeyboardShowing)
+				RestorePosition();
+
+			IsKeyboardShowing = false;
+			View = null;
+			LastScrollView = null;
+			KeyboardFrame = CGRect.Empty;
+			StartingContentInsets = new UIEdgeInsets();
+			StartingScrollIndicatorInsets = new UIEdgeInsets();
+			StartingContentInsets = new UIEdgeInsets();
+		});
+	}
+
+	internal static void Destroy()
+	{
+		if (WillShowToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(WillShowToken);
+		if (DidHideToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(DidHideToken);
+		if (TextFieldToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(TextFieldToken);
+		if (TextViewToken is not null)
+			NSNotificationCenter.DefaultCenter.RemoveObserver(TextViewToken);
+	}
+
+	// Used to get the numeric values from the UserInfo dictionary's NSObject value to CGRect.
+	// Doing manually since CGRectFromString is not yet bound
+	static CGRect? DescriptionToCGRect(string? description)
+	{
+		// example of passed in description: "NSRect: {{0, 586}, {430, 346}}"
+
+		if (description is null)
+			return null;
+
+		// remove everything except for numbers and commas
+		var temp = Regex.Replace(description, @"[^0-9,]", "");
+		var dimensions = temp.Split(',');
+
+		if (int.TryParse(dimensions[0], out var x) && int.TryParse(dimensions[1], out var y)
+			&& int.TryParse(dimensions[2], out var width) && int.TryParse(dimensions[3], out var height))
+		{
+			return new CGRect(x, y, width, height);
+		}
+
+		return null;
+	}
+
+	static async Task SetUpTextEdit()
+	{
+		if (View is null)
+			return;
+
+		CursorRect = null;
+
+		RootController = View.FindResponder<ContainerViewController>()?.View;
+
+		// the cursor needs a small amount of time to update the position
+		await Task.Delay(5);
+
+		CGRect? localCursor = null;
+
+		if (View is IUITextInput textInput)
+		{
+			var selectedTextRange = textInput.SelectedTextRange;
+			if (selectedTextRange is UITextRange selectedRange)
+			{
+				localCursor = textInput.GetCaretRectForPosition(selectedRange.Start);
+				if (localCursor is CGRect local)
+					CursorRect = View.ConvertRectToView(local, null);
+			}
+		}
+
+		TextViewTopDistance = localCursor is CGRect cGRect ? 20 + (int)cGRect.Height : 20;
+
+		await AdjustPositionDebounce();
+	}
+
+	// Used to debounce calls from different oberservers so we can be sure
+	// all the fields are updated before calling AdjustPostition()
+	internal static async Task AdjustPositionDebounce()
+	{
+		DebounceCount++;
```

Comment:
Not totally necessary because this will most likely be called by the UIThread but can we use `Interlocked.Increment` here for thread safety?

Reply:
better safe than sorry